src/pages/rekey-reason/cat-c/__tests__/rekey-reason.cat-c.page.spec.ts

```diff
@@ -0,0 +1,247 @@
+import { NavControllerMock, PlatformMock, ConfigMock, LoadingControllerMock, ModalControllerMock } from 'ionic-mocks';
+import { RekeyReasonCatCPage } from '../rekey-reason.cat-c.page';
+import { AuthenticationProvider } from '../../../../providers/authentication/authentication';
+import { AuthenticationProviderMock } from '../../../../providers/authentication/__mocks__/authentication.mock';
+import { Store, StoreModule } from '@ngrx/store';
+import { AppModule } from '../../../../app/app.module';
+import { RekeyReasonModel } from '../../rekey-reason.model';
+import { getUploadStatus } from '../../rekey-reason.selector';
+import {
+  SendCurrentTest,
+  SendCurrentTestSuccess,
+  SendCurrentTestFailure,
+} from '../../../../modules/tests/tests.actions';
+import { rekeyReasonReducer } from '../../rekey-reason.reducer';
+import { CAT_C } from '../../../page-names.constants';
+import { AppInfoModel } from '../../../../modules/app-info/app-info.model';
+import {
+  IpadIssueSelected,
+  OtherSelected,
+  OtherReasonUpdated,
+  TransferSelected,
+  IpadIssueLostSelected,
+} from '../../../../modules/tests/rekey-reason/rekey-reason.actions';
+import { By } from '@angular/platform-browser';
+import { SetExaminerConducted } from '../../../../modules/tests/examiner-conducted/examiner-conducted.actions';
+import { MockComponent } from 'ng-mocks';
+import { FindUserProvider } from '../../../../providers/find-user/find-user';
+import { FindUserProviderMock } from '../../../../providers/find-user/__mocks__/find-user.mock';
+import { IpadIssueComponent } from '../../components/ipad-issue/ipad-issue';
+import { TransferComponent } from '../../components/transfer/transfer';
+import { OtherReasonComponent } from '../../components/other-reason/other-reason';
+import { NavigationStateProvider } from '../../../../providers/navigation-state/navigation-state';
+import { NavigationStateProviderMock } from '../../../../providers/navigation-state/__mocks__/navigation-state.mock';
+
+describe('RekeyReasonCatCPage', () => {
+  let fixture: ComponentFixture<RekeyReasonCatCPage>;
+  let component: RekeyReasonCatCPage;
+  let loadingController: LoadingController;
+  let navContoller: NavController;
+  let modalController: ModalController;
+  let store$: Store<AppInfoModel>;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [
+        RekeyReasonCatCPage,
+        MockComponent(IpadIssueComponent),
+        MockComponent(TransferComponent),
+        MockComponent(OtherReasonComponent),
+      ],
+      imports: [
+        IonicModule,
+        StoreModule.forRoot({
+          journal: () => ({
+            isLoading: false,
+            lastRefreshed: null,
+            slots: {},
+            selectedDate: '',
+            examiner: {
+              staffNumber: '1234567',
+            },
+          }),
+          tests: () => ({
```

Comment:
TODO to mention that this will be updated to use Cat C schema test when available.

Reply:
Just added :)